File: 1. Two Sum/Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result(2, 0);
        unordered_map<int, int> hash;
        int nLength = nums.size();
        int remain;
        
        for(int i = 0; i < nLength; i++){
            remain = target - nums[i];
            if(hash.count(remain) != 0){
                result[0] = i;
                result[1] = hash[remain];
                break;
            }
            //This is how unordered_map insert new element
            hash[nums[i]] = i;
            
        }
        
        return result;
    }
};

File: Java/1. Two Sum/Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> recordSet = new HashMap<Integer, Integer>();
        int[] result = new int[2];
        recordSet.put(nums[0], 0);
        for(int i = 1; i < nums.length; i++){
            int remain = target - nums[i];
            if(recordSet.containsKey(remain)){
                result[0] = i;
                result[1] = recordSet.get(remain);
                break;
            }
            recordSet.put(nums[i], i);
        }
        
        return result;
    }
}